src/Service/Storage/Ef/DbGlobalContext.cs

```diff
@@ -62,8 +62,14 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
                 .IsRequired();
         });
 
-        modelBuilder.Entity<ApplicationEvent>()
-            .HasKey(x => x.Id);
+        modelBuilder.Entity<ApplicationEvent>(builder =>
+        {
+            builder.HasKey(x => x.Id);
+            builder.HasOne(x => x.Application)
+                .WithMany(x => x.Events)
+                .HasForeignKey(x => x.TenantId)
+                .IsRequired();
+        });
```

Comment:
@jrmccannon This seems like we need migrations?

Reply:
Done